enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,407 @@
+        mode: 0644
+        overwrite: true
+        path: /etc/nmstate/openshift/worker-1.yaml
+```
+
+The filenames must correspond to the hostname of each node as that will be
+used to map the configuration to the node. A separate manifest will need
+to be provided for masters and workers, which is typical when working with
+machine-configs.
+
+When the new service runs, it will copy the appropriate YAML file (based on
+the hostname of the node) to /etc/nmstate so it will be applied by the normal
+system NMState service. If there is already a file in /etc/nmstate with the
+same name, the service will fail in order to avoid overwriting important
+system configuration.
+
+Alternatively, there will be a mechanism to deploy a single cluster-wide
+configuration. That might look like:
+
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: master
+  name: 10-br-ex-master
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,aW50ZXJmYWNlczoKLSBuYW1lOiBlbnAyczAKICB0eXBlOiBldGhlcm5ldAogIHN0YXRlOiB1cAogIGlwdjQ6CiAgICBlbmFibGVkOiBmYWxzZQogIGlwdjY6CiAgICBlbmFibGVkOiBmYWxzZQotIG5hbWU6IGJyLWV4CiAgdHlwZTogb3ZzLWJyaWRnZQogIHN0YXRlOiB1cAogIGNvcHktbWFjLWZyb206IGVucDJzMAogIGlwdjQ6CiAgICBlbmFibGVkOiBmYWxzZQogICAgZGhjcDogZmFsc2UKICBpcHY2OgogICAgZW5hYmxlZDogZmFsc2UKICAgIGRoY3A6IGZhbHNlCiAgYnJpZGdlOgogICAgcG9ydDoKICAgIC0gbmFtZTogZW5wMnMwCiAgICAtIG5hbWU6IGJyLWV4Ci0gbmFtZTogYnItZXgKICB0eXBlOiBvdnMtaW50ZXJmYWNlCiAgc3RhdGU6IHVwCiAgaXB2NDoKICAgIGVuYWJsZWQ6IHRydWUKICAgIGRoY3A6IHRydWUKICBpcHY2OgogICAgZW5hYmxlZDogZmFsc2UKICAgIGRoY3A6IGZhbHNl
+        mode: 0644
+        overwrite: true
+        path: /etc/nmstate/openshift/cluster.yaml
+```
+
+The only difference is there will be a single file applied to each node of the
+specified role. The name of the file must be `cluster.yaml`.
+If there are both node-specific and cluster-wide configuration
+files present, the node-specific one will take precedence. Again, separate
+manifests will be needed for each role because of how MCO works.
+
+When this mechanism is in use, a flag will be set to indicate to configure-ovs
+that it should not attempt to configure br-ex.
+
+Day 2 changes will be handled by Kubernetes-NMState. For the initial
```

Comment:
Two questions here:

- Do we prevent in any way processing day-2 changes on the MachineConfigs described in this procedure? If not, would that be an alternative mechanism to deploy day-2 changes if user is willing to put up with a reboot? I guess the answer is yes, to allow for the scale out scenario.
- Do we prevent day-2 with Kubernetes-NMState to deployments that are still relying on configure-ovs

Reply:
- A reboot won't happen on day 2 changes to these files because we are adding the /etc/nmstate/openshift path to the reboot-less update list in MCO. We won't otherwise prevent changes to these files, but changes to existing nodes will have no effect because these configs are only applied on first boot.
- We have a warning in the kubernetes-nmstate docs that modification of br-ex and the underlying interface is not allowed. We'll need to modify that to allow changes if this new feature is used instead of configure ovs.